### PR TITLE
Overhaul TLS and ingress certificates docs

### PR DESCRIPTION
**Pull request by mamachanko**
* Adds an inventory of components, how they use ingress+TLS
* Mentions wildcard certificates and their second-class support
* Improves structure and presentation
* Adds more specific examples of ingress issuer configurations
* Refers from the hardening guide

---
### Target

Targeted for TAP `1.5`

### Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

None

### CODE REVIEW

File: security-and-compliance/tls-and-certificates/ingress/issuer.hbs.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    #! my-tap-values.yaml
+    #! ...
+    shared:
+      ingress_issuer: my-company-ca
+    #! ...
+    ```
+
+1. Apply TAP's installation values
+
+Once the configuration is applied, components eventually obtain certificates
```
Comment (Petahhh) [inline security-and-compliance/tls-and-certificates/ingress/issuer.hbs.md:250]: Can you briefly describe how one would conveniently verify the certificates have been updated? It'll be helpful  but I don't think it's absolutely necessary. I find it personally helpful at the end of a procedure to have some hints if I completed it correctly

**Review by Petahhh — APPROVED** [review]
Thank you for consolidating all this information! It was clear and easy to follow.

One thing I worry about (and this is not particular to you but maybe a system doc maintenance challenge) is how we ensure that big table is accurately maintained. This is not a blocking ask nor do I need a response but something I thought I'd highlight.

**Comment by mamachanko** [comment]
Thank you @Petahhh !

> One thing I worry about (and this is not particular to you but maybe a system doc maintenance challenge) is how we ensure that big table is accurately maintained. 

You are right in calling out the risk of documentation drift. My current strategy is: it's maintained by me / my team and we create awareness by sharing edits with the respective teams.

**Comment by kcoriordan** [comment]
Hi @mamachanko 
here is the link to the updated table, let me know what you think: https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.6/tap/security-and-compliance-tls-and-certificates-ingress-inventory.html

File: security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+>**Note** The lowercase "ingress" refers to any resource which facilitates
+>ingress, e.g. core `Ingress` and Contour's `HTTPProxy`.
+
+To help with planning and accounting of TLS certificates, here's an inventory
+of TAP's components respective ingress:
+
+Package name | Profiles | Has ingress | Ingress purpose | Supports ingress issuer | Supports wildcards | # of ingress | SANs | TLS Documentation
+---|---|---|---|---|---|---|---|---|
+accelerator.apps.tanzu.vmware.com | full, light, view | | | | | | | |
+api-portal.tanzu.vmware.com | full, view | ✅ | Serves the API portal | |  | `1` | `api-portal.<ingress-domain>` (configurable) | [docs](../../../api-portal/about.hbs.md)
```
[the remark this answers is not in the document]
Comment (mamachanko) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:22]: Fair point. They are, one way or another, all configurable, `sso.apps.tanzu.vmware.com` included. The components generally break up into two camps:
* Those which install a single ingress resource in the form of `<component>.<domain_name>`, e.g. TAP GUI
* Those which install an ingress resource per API instance which gets templated from a `domain_template` feeding `<domain_name>`, e.g. `cnrs.tanzu.vmware.com` and `sso.apps.tanzu.vmware.com`

[the remark this answers is not in the document]
Comment (mamachanko) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:22]: Exactly! Do you think that's good or bad? Do you reckon saying _"no"_ or ❌ out loud would help?

File: security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+endpoints and those which do. The latter further break down into those which
+have a fixed number of ingress endpoints and those which offer Kubernetes APIs
+with ingress.
+
+>**Note** The lowercase "ingress" refers to any resource which facilitates
+>ingress, e.g. core `Ingress` and Contour's `HTTPProxy`.
+
+To help with planning and accounting of TLS certificates, here's an inventory
+of TAP's components respective ingress:
+
```
[the remark this answers is not in the document]
Comment (mamachanko) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:18]: At first I was sceptical about trimming away components with no ingress, what with it be _The Inventory_. Then it grew on me and I think you are making really good points! Thank you very much for digging into this.

File: security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kcoriordan) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:22]: I added this as a note at the bottom of the table._

File: security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+apiserver.appliveview.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+app-scanning.apps.tanzu.vmware.com  | | | | | | | | |
+application-configuration-service.tanzu.vmware.com | | | | | | | | |
+backend.appliveview.tanzu.vmware.com | full, light, view | | | | | | | |
+bitnami.services.tanzu.vmware.com | full, iterate, run | | | | | | | |
+buildservice.tanzu.vmware.com | full, light, iterate, build, buildservice | | | | | | | |
+carbonblack.scanning.apps.tanzu.vmware.com |  | | | | | | | |
+cartographer.tanzu.vmware.com | full, light, iterate, build, run | | | | | | | |
+cert-manager.tanzu.vmware.com | full, light, iterate, build, run, view | | | | | | | |
+cnrs.tanzu.vmware.com | full, light, iterate, run | ✅ | Instances of Knative's `Service` have ingress | ✅ | ✅ | `# of Services` | SANs depend on the component's `domain_template` (configurable) | [docs](../../../cloud-native-runtimes/about.hbs.md)
```
[the remark this answers is not in the document]
Comment (kcoriordan) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:33]: I wasn't sure tbh, so I left it as is.

File: security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ootb-supply-chain-testing-scanning.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+ootb-supply-chain-testing.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+ootb-templates.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+policy.apps.tanzu.vmware.com | full, iterate, run | | | | | | | |
+scanning.apps.tanzu.vmware.com | full, build | | | | | | | |
+service-bindings.labs.vmware.com | full, light, iterate, run | | | | | | | |
+services-toolkit.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+snyk.scanning.apps.tanzu.vmware.com | | | | | | | | |
+spring-boot-conventions.tanzu.vmware.com | full, light, iterate, build | | | | | | | |
+spring-cloud-gateway.tanzu.vmware.com | | ✅ | Instances of [SpringCloudGateway](../../../spring-cloud-gateway/about.hbs.md) have ingress | | ✅ | `# of SpringCloudGateway` | configurable | [docs](../../../spring-cloud-gateway/about.hbs.md) |
```
[the remark this answers is not in the document]
Comment (kcoriordan) [reply security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:58]: I added a link to this.

**Comment by kcoriordan** [comment]
OK, made a few more updates not, I think it looks good, can you take another look to make sure you're happy with it
https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.6/tap/security-and-compliance-tls-and-certificates-ingress-inventory.html
@mamachanko

**Comment by mamachanko** [comment]
> thanks Max @mamachanko - I can't change the background color unfortunately I made a few other changes based on a team edit review, to bring in line with style guide, accessibility, and translation standards:
> 
> -For the # of Ingress column – I changed that to 'number of ingress'. And had to make the items plural as a result. Can you check that's ok, and also, does this content need to be in code back ticks - I think they could be removed and it can be in normal text? I changed the variable values from <> to caps with dashes I changed the x and tick to Yes and No

@kcoriordan 👍🏻  to all your changes!

> does this content need to be in code back ticks - I think they could be removed and it can be in normal text?

💯 agreed!

**Pull request merged**